openquake/hazardlib/gsim/megawati_2003.py

```diff
@@ -0,0 +1,181 @@
+    #: Sumatra strike-slip fault
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are spectral acceleration,
+    #: peak ground veloacity and peak ground acceleration
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        PGV,
+        SA
+    ])
+
+    #: Supported intensity measure component is geometric mean
+    #: of two horizontal components,
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Supported standard deviation types is total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL
+    ])
+
+    #: No site parameter required. This GMPE is for very hard rock conditions
+    REQUIRES_SITES_PARAMETERS = set(())
+
+    #: Required rupture parameter is magnitude
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: Required distance measure is hypocentral distance, and azimuth
+    REQUIRES_DISTANCES = set(('rhypo', 'azimuth'))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # Check that the GSIM supports the standard deviations requested
+        assert all(stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+                   for stddev_type in stddev_types)
+        coe = self.COEFFS[imt]
+        mean = (self._get_magnitude_scaling(coe, rup.mag) +
+                self._get_distance_scaling(coe, dists.rhypo) +
+                self._get_azimuth_correction(coe, dists.azimuth))
+        # Convert to g
+        if isinstance(imt, (PGA, SA)):
+            mean = np.log(np.exp(mean) / (100.0 * g))
+        # Compute std
+        stddevs = self._compute_std(coe, stddev_types, len(dists.azimuth))
+        return mean, stddevs
+
+    def _get_magnitude_scaling(self, coe, mag):
+        """
+        Returns the magnitude scaling term
+        """
+        return coe["a0"] + coe["a1"] * mag + coe["a2"] * mag**2.
+
+    def _get_distance_scaling(self, coe, rhypo):
+        """
+        Returns the distance scaling term
+        """
+        return coe["a3"] * np.log(rhypo) + coe["a4"] * rhypo
+
+    def _get_azimuth_correction(self, coe, azimuth):
+        """
+        This is the azimuth correction defined in the functional form (see
+        equation 3 at page 2256)
+        """
+        term1 = abs(np.cos(np.radians(2*azimuth)))
+        term2 = abs(np.sin(np.radians(2*azimuth)))*coe['a5']
+        aa = np.log(np.max(np.hstack((term1, term2))))
+        return aa
+
+    def _compute_std(self, coe, stddev_types, num_sites):
+        """
+        Returns the total standard deviation according to table III at page
+        2257. Table obtained via http://www.onlineocr.net/
+        """
+        return [np.ones_like((num_sites))*coe['sigma']]
```

Comment:
This is totally different from the usual way we handle the standard deviations. Also, I'm not sure this would work in a Shakemap application when 2D arrays are needed.